packages/tax/src/services/tax-module-service.ts

```diff
@@ -278,6 +267,38 @@ export default class TaxModuleService<
     return toReturn.flat()
   }
 
+  private prepareTaxRegionInputForCreate(
+    data: TaxTypes.CreateTaxRegionDTO | TaxTypes.CreateTaxRegionDTO[]
+  ) {
+    const input = Array.isArray(data) ? data : [data]
+    return input.reduce(
```

Comment:
This code is really difficult to read and understand. Maybe we can use a simple loop instead (usually long `reduce` functions make code less readable), and try to structure the logic a bit better?